airflow/models/dag.py

```diff
@@ -611,12 +614,18 @@ def __init__(
         self.last_loaded: datetime = timezone.utcnow()
         self.safe_dag_id = dag_id.replace(".", "__dot__")
         self.max_active_runs = max_active_runs
+        self.max_failure_runs = max_failure_runs
         if self.timetable.active_runs_limit is not None:
             if self.timetable.active_runs_limit < self.max_active_runs:
                 raise AirflowException(
                     f"Invalid max_active_runs: {type(self.timetable)} "
                     f"requires max_active_runs <= {self.timetable.active_runs_limit}"
                 )
+        if self.max_failure_runs is not None and self.max_failure_runs < 0:
+            raise AirflowException(
+                f"Invalid max_failure_runs: {str(self.max_failure_runs)}"
+                f"requires max_failure_runs >= 0"
```

Comment:
```suggestion
                f"Invalid max_failure_runs: {self.max_failure_runs}. "
                "Requires max_failure_runs >= 0"
```

Don't believe string conversion is required here. Also adding whitespace between sentences + remove unnecessary f-string.